lms/djangoapps/program_enrollments/api/writing.py

```diff
@@ -424,3 +454,51 @@ def _organize_requests_by_external_key(enrollment_requests):
             continue
         requests_by_key[key] = request
     return requests_by_key, duplicated_keys
+
+
+def _get_conflicting_active_course_enrollments(
+    requests_by_key,
+    existing_course_enrollments,
+    program_uuid,
+    course_key
+):
+    """
+    Process the list of existing course enrollments together with
+    the enrollment request list stored in 'requests_by_key'. Detect
+    whether we have conflicting ACTIVE ProgramCourseEnrollment entries.
+    When detected, log about it and return the conflicting entry with
+    duplicated status.
+
+    Arguments:
+        requests_by_key (dict)
+        existing_course_enrollments (queryset[ProgramCourseEnrollment]),
+        program_uuid (UUID|str),
+        course_key (str)
+
+    Returns:
+        results (dict) with detected conflict entry, or empty dict.
+    """
+    conflicted_by_user_key = {}
+
+    requested_statuses_by_user_key = {
+        key: request.get('status') for key, request in requests_by_key.items()
+    }
+
+    for existing_enrollment in existing_course_enrollments:
+        external_user_key = existing_enrollment.program_enrollment.external_user_key
+        requested_status = requested_statuses_by_user_key.get(
+            existing_enrollment.program_enrollment.external_user_key
+        )
+        if requested_status and \
```

Comment:
```suggestion
        if (requested_status 
            and requested_status == ProgramCourseEnrollmentStatuses.ACTIVE
            and existing_enrollment.is_active
            and str(existing_enrollment.program_enrollment.program_uuid) != str(program_uuid)):
```

a potentially-more-readable way to write this